awsutil/generate_credentials.go

```diff
@@ -230,6 +234,22 @@ func (c *CredentialsConfig) GenerateCredentialChain(opt ...Option) (*credentials
 				// Add the web identity role credential provider
 				providers = append(providers, webIdentityProvider)
 			}
+		} else if c.WebIdentityToken != "" {
+			c.log(hclog.Debug, "adding web identity provider with token", "roleARN", c.RoleARN)
+			sess, err := session.NewSession()
+			if err != nil {
+				return nil, errors.Wrap(err, "error creating a new session to create a WebIdentityRoleProvider with token")
+			}
+			webIdentityProvider := stscreds.NewWebIdentityRoleProviderWithToken(sts.New(sess), c.RoleARN, c.RoleSessionName, FetchTokenContents(c.WebIdentityToken))
+
+			// Check if the webIdentityProvider can successfully retrieve
+			// credentials (via sts:AssumeRole), and warn if there's a problem.
+			if _, err := webIdentityProvider.Retrieve(); err != nil {
```

Comment:
I see that `RetrieveCreds()` calls `GenerateCredentialChain()` every time, so does this mean we are doubling the number of credentials generated?

Reply:
I like that we do this for consistency with the `WebIdentityTokenFile` option, but perhaps it would be nice to be able to optionally turn off this validation check. Maybe in a follow-on PR though to keep this one tightly scoped?

Reply:
Given the fact that we are using this library in VSO, I feel like we should definitely provide an opt out from this sort of validation. The validation seems more geared towards the vault CLI use-case.

Reply:
Ok, I tried adding a WithSkipWebIdentityValidity option in 452aa5a6f488f459043678674fb24774e25ff9cf to skip the extra Retrieve() with the webIdentityProvider's. The AssumeRoleProvider looks like its doing some caching under the hood, so that's why the option is scoped to just the webIdentityProvider's.

Reply:
I suspect the main reason the extra Retrieve() is called here is to help with debugging, since otherwise the aws sdk [doesn't give back error messages from failed providers](https://github.com/aws/aws-sdk-go/blob/v1.34.0/aws/credentials/chain_provider.go#L78) as it goes down the credentials chain if a subsequent provider succeeds. (Just something to keep in mind.)